### ISSUE TITLE
Include experiment name in last checkpoint and export loss for k-best

### ISSUE THREAD

**Issue by melisande-c** [CAREamics/careamics#792]
## Problem
1. If multiple experiments are run in the same working directory, the last checkpoints will be saved as `"last.ckpt"` `"last-v1.ckpt"` etc. this makes it hard to know which experiment the checkpoint was from. This was encountered as a problem in https://github.com/CAREamics/careamics/issues/779#issuecomment-3942495862.

2. Additionally, with the k-best checkpoints they are saved only with the number of steps and the epoch, but no way to know which had the best validation loss.

## Potential solution
1. We can set the name of the last checkpoint to include the experiment name as documented in the lightning docs https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.callbacks.ModelCheckpoint.html:

    it has to be done through setting the attribute
    ```python
    checkpoint_callback.CHECKPOINT_NAME_LAST = f"{experiment_name}-last"
    ```

2. The validation loss of the k-best checkpoints could be included in the file name, or at the end of training we could save the data from `ModelCheckpoint` in a file.
